### PR TITLE
HDDS-4327. Potential resource leakage using BatchOperation.

### PR DESCRIPTION
**Pull request by bharatviswa504**
## What changes were proposed in this pull request?

Potential resource leakage using BatchOperation.

Use try enclosed resource/close the batch once after it usage is completed.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-4327

## How was this patch tested?

Existing tests.

### CODE REVIEW

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/SCMContainerManager.java [hunk 1]
```diff
@@ -397,33 +399,29 @@ public void updateDeleteTransactionId(Map<Long, Long> deleteTransactionMap)
...
       return;
     }
-    org.apache.hadoop.hdds.utils.db.BatchOperation batchOperation =
-        batchHandler.initBatchOperation();
-    lock.lock();
-    try {
-      for (Map.Entry<Long, Long> entry : deleteTransactionMap.entrySet()) {
+    try(org.apache.hadoop.hdds.utils.db.BatchOperation batchOperation =
+        batchHandler.initBatchOperation()) {
+      lock.lock();
```
Comment (hanishakoneru) [inline hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/SCMContainerManager.java:404]: Shouldn't lock() be called outside the try block? If there is a chance that the initBatchOperation() can fail, then lock.unlock() would throw IllegalMonitorStateException as the lock is not held by it.

**Review by hanishakoneru — COMMENTED** [review]
Thanks @bharatviswa504 for fixing this. 
Have one question (posted inline). LGTM otherwise.

File: hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/SCMContainerManager.java (hunk 1 above)
Comment (bharatviswa504) [reply hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/SCMContainerManager.java]: Good catch. Fixed in the latest commit.

**Comment by hanishakoneru** [comment]
Thanks @bharatviswa504. +1 for merge.

**Pull request merged**